workers/unity/Assets/Gdk/Core/Components/WorldCommandsTranslation.cs

```diff
@@ -285,53 +308,56 @@ public override void RegisterWithDispatcher(Dispatcher dispatcher)
 
         public void OnCreateEntityResponse(CreateEntityResponseOp op)
         {
+            CreateEntityRequest request = requestIdToCreateEntityRequest[op.RequestId.Id];
+            requestIdToCreateEntityRequest.Remove(op.RequestId.Id);
+
+            var response =
+                new CreateEntityResponse((CommandStatusCode) op.StatusCode, op.Message, op.EntityId.Value.Id, request);
```

Comment:
Any particular reason for the move up? It seems unnecessary to compute `response` in case the check fails and we return.